packages/netlify-cms-core/src/actions/config.js

```diff
@@ -80,7 +103,7 @@ export function applyDefaults(config) {
             collection = setDefaultPublicFolder(collection);
             collection = collection.set(
               'fields',
-              traverseFields(collection.get('fields'), setDefaultPublicFolder),
+              traverseFields(collection.get('fields'), setDefaultPublicFolder, setSnakeCaseConfig),
```

Comment:
Do we need an additional argument here?
The second argument is a function to update the field.
We could combine `setDefaultPublicFolder` and `setSnakeCaseConfig` into a single function, however I think it might be better to re-run `traverseFields` with `setSnakeCaseConfig` outside and before of `applyDefaults`.
Maybe we could call it `normalizeConfig`, then `applyDefaults` can only deal with snake casing.
That would make it easier to remove the migration logic once we deprecate the old casing.

Reply:
Okay, I would create a separate function for this. I thought rerunning `traverseFields` would be an expensive operation. But if that isn't the case, then yes, separating it would be better